config/validation.go

```diff
@@ -6,7 +6,7 @@ func ValidateConfig(c Config) error {
 	err := ValidateConfigApi(c)
 	if err != nil {
 		return err
-	} else if (c.AccessToken == "" || c.AccessToken == "revoked") && c.ClientID == "" {
+	} else if (c.AccessToken == "" || c.AccessToken == "revoked") && c.ClientID == "" && c.ClientCertPath == "" {
 		return errors.NewRevokedTokenError()
```

Comment:
I decided to rename this to `errors.NewUnauthenticatedError`, and left the error message as "You are not currently authenticated. Please log in to continue." The message is wrong if the user has actually tried/failed to supply a client certificate… but honestly I think that's an advanced feature and anyone using it can figure things out.

Reply:
I am wondering why you say
> The message is wrong if the user has actually tried/failed to supply a client certificate…

It seems to me that even in this case the message would be correct: you would need to provide a valid certificate in order to be logged in. Or are you saying that providing an mTLS certificate doesn't qualify as "logging in"?